core/primitives/src/stateless_validation.rs

```diff
@@ -219,6 +219,24 @@ pub struct StoredChunkStateTransitionData {
     pub receipts_hash: CryptoHash,
 }
 
+#[derive(Debug)]
+pub struct EndorsementStats {
+    pub total_stake: Balance,
+    pub endorsed_stake: Balance,
+    pub total_validators_count: usize,
+    pub endorsed_validators_count: usize,
+}
+
+impl EndorsementStats {
+    pub fn has_enough_stake(&self) -> bool {
+        self.endorsed_stake > self.required_stake()
```

Comment:
I remember dealing with a scenario in tests where required_stake was zero due to null validator assignment. Not sure where that change disappeared. `endorsement_stats.endorsed_stake as f64 / endorsement_stats.total_stake as f64` might die due to division by zero

Reply:
having `total_stake == 0` in case of block producer doesn't seem possible with a correct setup, and since current tests are passing I suggest to keep it as it is without `total_stake == 0` check

Reply:
Agree, but maybe like a quick assert or something might be a good idea during initialization. But I'll leave it to you if you find it nice to have! :)

Reply:
f64/f64 won't ever die, it might return inf or nan, and the metric would be messed up, other than that it's not a big deal.

Reply:
Aah, this isn't C++ 😆